ninja.lua

```diff
@@ -269,7 +285,7 @@ function ninja.generateProjectCfg(cfg)
 		elseif path.iscppfile(node.abspath) then
 			objfilename = obj_dir .. "/" .. node.objname .. intermediateExt(cfg, "cxx")
 			objfiles[#objfiles + 1] = objfilename
-			srcpath = project.getrelative(cfg.workspace, cfg.location .. "/" .. node.relpath)
+			srcpath = project.getrelative(cfg.workspace, "/" .. node.relpath)
```

Comment:
can we have just srcpath = node.abspath ? I can't see any reason why we wouldn't want it to be the absolute path.

I get something like this:
build contrib/libzip/build/obj/Release/mkstemp.o: cc /home/rui/dev/premake/contrib/libzip/mkstemp.c

so probably will always works :) works for premake and my project at least :P